components/camel-file-watch/src/main/java/org/apache/camel/component/file/watch/FileWatchEndpoint.java

```diff
@@ -0,0 +1,211 @@
+/*
+ * Licensed to the Apache Software Foundation (ASF) under one or more
+ * contributor license agreements.  See the NOTICE file distributed with
+ * this work for additional information regarding copyright ownership.
+ * The ASF licenses this file to You under the Apache License, Version 2.0
+ * (the "License"); you may not use this file except in compliance with
+ * the License.  You may obtain a copy of the License at
+ *
+ *      http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package org.apache.camel.component.file.watch;
+
+import java.util.Arrays;
+import java.util.HashSet;
+import java.util.Set;
+
+import io.methvin.watcher.hashing.FileHasher;
+import org.apache.camel.Consumer;
+import org.apache.camel.MultipleConsumersSupport;
+import org.apache.camel.Processor;
+import org.apache.camel.Producer;
+import org.apache.camel.component.file.watch.constants.FileEventEnum;
+import org.apache.camel.spi.Metadata;
+import org.apache.camel.spi.UriEndpoint;
+import org.apache.camel.spi.UriParam;
+import org.apache.camel.spi.UriPath;
+import org.apache.camel.support.DefaultEndpoint;
+
+/**
+ * The file-watch is used to monitor file events in directory using {@link java.nio.file.WatchService}
+ */
+@UriEndpoint(firstVersion = "3.0.0", scheme = "file-watch", title = "file-watch", syntax = "file-watch:path",
+label = "file", consumerOnly = true)
+public class FileWatchEndpoint extends DefaultEndpoint implements MultipleConsumersSupport {
+    @UriPath(label = "consumer", description = "Path of directory to consume events from.")
+    @Metadata(required = true)
+    private String path;
+
+    @UriParam(label = "consumer",
+    description = "Coma separated list of events to watch. Allowed values are: CREATE, MODIFY, DELETE.",
```

Comment:
There is an attribute to set the enum values separated by comma - i think its `enums`

Reply:
Thanks, added.